graphstorm-processing/tests/test_repartition_files.py

```diff
@@ -396,6 +410,33 @@ def mock_json_load(file):
                     expected_counts, edge_feature_dict["data"]
                 ):
                     absolute_feature_filepath = os.path.join(TEMP_DATA_PREFIX, feature_filepath)
+                    filemeta = pq.read_metadata(absolute_feature_filepath)
+                    file_rows = filemeta.num_rows
                     assert (
-                        expected_count == pq.read_metadata(absolute_feature_filepath).num_rows
+                        expected_count == file_rows
                     ), f"Count mismatch for {feature_name}, {absolute_feature_filepath}"
+                    # Ensure the flat array has the correct metadata embedded
+                    if feature_name == "label" and task_type not in {
+                        "link_predict",
+                        "link_prediction",
```

Comment:
Why do we have two kinds of labels for link prediction here? ( Feel free to correct me if there is anything wrong

Reply:
Needed for legacy with NeptuneML